include/roboteam_utils/Line.h

```diff
@@ -182,6 +190,16 @@ class Line {
          */
         [[nodiscard]] bool doesIntersect(const LineSegment &line) const;
 
+    private:
+        /**
+         * Get the intersection point between this line and the given line as parameter. Moreover compute how much the given line is extended relatively in the direction of
+         * the start of the line towards the end of the line to find the intersection point.
+         *
+         * @param line Line to check against.
+         * @return std::nullopt if the lines do not intersect. Otherwise returns the intersection point and the relative extension of the given line to reach the intersection
+         * point.
+         */
+        [[nodiscard]] std::optional<std::pair<Vector2, float>> generalIntersect(const Line &line) const;
```

Comment:
I am personally not a fan of this. We have disagreed on this in the past but I think it's better to keep code decoupled for important math functions like these. Now you are coupling all of the line intersections with eachother when some use different things. Yes it's 30 lines less code in our codebase, but it also makes it harder for people to keep an overview of the math calculations and harder for people to edit and understand them IMO. Also by using std::pair you are decreasing the performance of these functions which are EXTREMELY performance important, actually.

Reply:
I do not agree, this is a code clone which immediately needs to be removed: 
- Most likely the code loses a lot of time because an algorithm is inefficiently implemented, this will only cause an insignificantly performance loss. Also bad design causes more developers to write inefficient code, because they do not understand the code. 
- Personally this is more understandable for me. With all code clones I had to find the differences between all of them and reason whether they all did what they are supposed to do. In this case you only have to check whether this function works correctly.

Reply:
Why did you even need to check the code clones? The whole point of function locality is that you only need to look at the function you have in front of you.

Reply:
Well you know if there is more code than there is more code that will be read, which causes additional development time. Developers not only look at the functions in front of them, they look at more functions. So removing code clones will reduce development time